services/management/commands/services_import/units.py

```diff
@@ -364,6 +364,17 @@ def _import_unit(syncher, keyword_handler, info, dept_syncher,
         obj_changed = True
         obj.geometry = obj.location
 
+    if obj.municipality_id is None and obj.location is not None:
+        div = AdministrativeDivisionGeometry.objects.select_related('division', 'division__type')\
+            .filter(boundary__contains=obj.location, division__type__type='muni')
+        if div:
+            muni_name = div[0].division.name_fi.lower()
+            muni = muni_by_name.get(muni_name)
+            municipality_id = muni.id
+        if municipality_id is not None:
+            obj.municipality_id = municipality_id
```

Comment:
By the way, you could change the above (lines 372 to 375) to just

    obj.municipality = muni_by_name.get(muni_name)

That's because in Django, you can set foreign keys by setting django models as attribute values, you don't need to use the ids (that's more low level). Additionally, there's no danger in setting obj.municipality to None, so you don't need to check for it separately.

Reply:
Of course, you would still need the muni is not None check for the

                LOGGER.info("Municipality_id added according to unit's location.")

call. (Since the log message is not true if the municipality is None.)